WCFDataService/Client/System/Data/Services/Client/Materialization/MaterializerEntry.cs

```diff
@@ -275,7 +275,7 @@ public static MaterializerEntry CreateEntryForLoadProperty(EntityDescriptor desc
         /// <returns>The materializer entry</returns>
         public static MaterializerEntry GetEntry(ODataEntry entry)
         {
-            return entry.GetAnnotation<MaterializerEntry>();
+            return entry == null ? null : entry.GetAnnotation<MaterializerEntry>();
```

Comment:
Seems DebugAssert would fail in ODataEntityMaterializer.ProjectionGetEntry if it returns null here. So consider to return null only when AutoNullPropagation is true?